### DISCUSSION TITLE
How is it determined whether a Rollout makes progress for the purposes of `progressDeadlineAbort`?

### DISCUSSION THREAD

**Discussion by andrii-korotkov-verkada** [argoproj/argo-rollouts#3798] [Q&A]
Is it based on making it to the next canary step or something else?
I'm checking my assumptions, since there's been a constant timing out of a particular Rollout with `progressDeadlineSeconds` set to 900, while the canary steps were supposed to take 5 min. There were 4 steps with 5 min pauses in between though.

**Comment by meeech** [comment]
my understanding is progressDeadline is when things dont 'happen'. Pauses don't count against it (to the best of my knowledge)

eg: we have a rollouts with progress deadline of 10 mins, but our steps look like this:
    # steps:
    #   - setWeight: 10
    #   - pause: { duration: 15m }
    #   - setWeight: 50
    #   - pause: { duration: 20m }
    #   - setWeight: 100
    #   - pause: { duration: 10m }

I've noticed we hit the progress deadline issue when rollouts would get 'stuck' where it couldnt scale up/down a replica set. 
And when we hit a bug (like https://github.com/argoproj/argo-rollouts/issues/3082)  

Hopefully someone can give you more nuance, since i'm sure there's some I'm not covering here.
